src/main/java/com/mraof/minestuck/data/recipe/MinestuckCombinationsProvider.java

```diff
@@ -502,7 +503,9 @@ protected void registerRecipes(Consumer<IFinishedRecipe> consumer)
 		CombinationRecipeBuilder.of(MSBlocks.GOLD_SEEDS).input(Items.WHEAT_SEEDS).and().namedInput(Items.GOLD_INGOT).build(consumer);
 		CombinationRecipeBuilder.of(MSItems.OBSIDIAN_BUCKET).input(Items.WATER_BUCKET).or().input(Items.LAVA_BUCKET).namedSource("buckets").build(consumer);
 		CombinationRecipeBuilder.of(MSItems.OBSIDIAN_BUCKET).input(Items.BUCKET).and().namedInput(Items.OBSIDIAN).build(consumer);
-		CombinationRecipeBuilder.of(MSItems.STONE_SLAB).input(Items.STONE).and().input(MSItems.CARVING_TOOL).build(consumer);
+		CombinationRecipeBuilder.of(MSItems.STONE_SLAB).input(Items.STONE).or().namedInput(MSItems.CARVING_TOOL).build(consumer);
+		CombinationRecipeBuilder.of(MSItems.STONE_SLAB).input(Items.STONE).and().namedInput(Items.WRITABLE_BOOK).build(consumer);
+		CombinationRecipeBuilder.of(MSItems.STONE_SLAB).input(Items.STONE).or().namedInput(MSItems.CAPTCHA_CARD).build(consumer);
```

Comment:
The logic behind the combinators makes it illogical for both `stone || x -> stone slab` and `stone && y -> stone slab` to exist. I'd suggest changing recipes to all use && or all use ||